src/deluge/io/midi/device_specific/midi_device_lumi_keys.h

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright (c) 2023 Aria Burrell (litui)
+ *
+ * This file is part of The Synthstrom Audible Deluge Firmware.
+ *
+ * The Synthstrom Audible Deluge Firmware is free software: you can redistribute it and/or modify it under the
+ * terms of the GNU General Public License as published by the Free Software Foundation,
+ * either version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program.
+ * If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "io/midi/midi_device.h"
+
+#define MIDI_DEVICE_LUMI_KEYS_VP_COUNT 1
+
+#define MIDI_DEVICE_LUMI_KEYS_SYSEX_START 0xF0
+#define MIDI_DEVICE_LUMI_KEYS_SYSEX_END 0xF7
+#define MIDI_DEVICE_LUMI_KEYS_SYSEX_SPACER 0x77
+#define MIDI_DEVICE_LUMI_KEYS_DEVICE 0x00 // All Devices
+
+#define MIDI_DEVICE_LUMI_KEYS_CONFIG_PREFIX 0x10
+#define MIDI_DEVICE_LUMI_KEYS_MPE_ZONE_PREFIX 0x00
+#define MIDI_DEVICE_LUMI_KEYS_MPE_CHANNELS_PREFIX 0x10
+#define MIDI_DEVICE_LUMI_KEYS_MIDI_MODE_PREFIX 0x20
+#define MIDI_DEVICE_LUMI_KEYS_KEY_PREFIX 0x30
+#define MIDI_DEVICE_LUMI_KEYS_KEY_COUNT 12
+#define MIDI_DEVICE_LUMI_KEYS_SCALE_PREFIX 0x60
+#define MIDI_DEVICE_LUMI_KEYS_SCALE_COUNT 19
+
+#define MIDI_DEVICE_LUMI_KEYS_CONFIG_ROOT_COLOUR_PREFIX 0x30
+#define MIDI_DEVICE_LUMI_KEYS_CONFIG_GLOBAL_COLOUR_PREFIX 0x20
+
+#define MIDI_DEVICE_LUMI_SCALE_MAJOR 0b101010110101
```

Comment:
It would be better if these get computed from the deluge scales - it looks to be a simple boolen bit array of which chromatic notes belong to the scale which can be derived easily from the deluge one

Reply:
Yes, that's exactly what it is, which is why it can't be computed from the Deluge one. These are used to /align/ to the Deluge scales (and abstract/arbitrary scales that may be enabled through adding a scale to a previously chromatic clip) because on the Lumi these are baked in presets (not as bit arrays... the bit arrays were my alternative to beefy uint8_t arrays such as those the deluge already uses).